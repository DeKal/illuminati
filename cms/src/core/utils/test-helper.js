import React from 'react'
import { shallow, mount } from 'enzyme'
import Theme from 'core/theme/ThemeWrapper'

export const shallowWithTheme = component => {
  const wrapper = shallow(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]').shallow()
}

export const mountWithTheme = component => {
  const wrapper = mount(
    <Theme>
      <div data-test-id="theme-wrapper">{component}</div>
    </Theme>
  )
  return wrapper.find('[data-test-id="theme-wrapper"]')
}
