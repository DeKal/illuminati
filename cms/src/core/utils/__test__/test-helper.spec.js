import * as React from 'react'
import { shallowWithTheme, mountWithTheme } from '../test-helper'

describe('Test helper', () => {
  it('should render shallowWithTheme with no error', function() {
    const wrap = shallowWithTheme(<div />)
    expect(wrap).toMatchSnapshot()
  })
  it('should render mountWithTheme with no error', function() {
    const wrap = mountWithTheme(<div />)
    expect(wrap).toMatchSnapshot()
  })
})
