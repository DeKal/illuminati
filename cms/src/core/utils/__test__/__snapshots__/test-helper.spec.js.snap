// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test helper should render mountWithTheme with no error 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <div />
</div>
`;

exports[`Test helper should render shallowWithTheme with no error 1`] = `
<div
  data-test-id="theme-wrapper"
>
  <div />
</div>
`;
